combat-trainer.lic

```diff
@@ -2000,32 +2003,38 @@ class TrainerProcess
   end
 
   def pickbox(game_state)
-    return if game_state.loaded
-    return if left_hand
-    return unless @pet_box_source
-    boxes = get_boxes(@pet_box_source)
-    return if boxes.empty?
-    box = boxes.first
-    # Check what's in the right hand so we can get it back
-    rhitem = @equipment_manager.item_by_desc(right_hand) if right_hand
-    @equipment_manager.stow_weapon("#{rhitem.adjective} #{rhitem.name}") if rhitem
-    return if right_hand # Couldn't empty right hand for some reason (typically a summoned weapon).  No picking allowed
+    return unless @pet_box_source && @pet_count > 0
+    return if game_state.loaded || game_state.weapon_is_summoned? || game_state.offhand?
+    return if left_hand && !game_state.currently_whirlwinding
+
+    box = get_boxes(@pet_box_source).first
+    return unless box
+
+    if right_hand
+      @equipment_manager.stow_weapon(game_state.weapon_name)
+      stowed_weapon = true
+    end
+    @equipment_manager.sheath_whirlwind_offhand if game_state.currently_whirlwinding
+   
     bput("get #{box} from my #{@pet_box_source}", 'You get', 'What were')
     retreat
-    response =  bput("pick my #{box} blind", 'not even locked', 'Roundtime', 'Find a more appropriate tool', 'You realize that would be next to impossible')
-    if response == 'not even locked'
-      # Dump the box so that combat-trainer lootprocess can take over
+    case bput("pick my #{box} blind", 'not even locked', 'Roundtime', 'Find a more appropriate tool', 'You realize that would be next to impossible')
+    when 'not even locked'
+      # Dump the box so that combat-trainer's lootprocess can take over
       bput("open my #{box}", 'You open', 'That is already open')
       2.times do
         bput("dismantle my #{box}", 'You can not dismantle', 'You dump the contents', 'You move your hands', 'You must be holding the object', 'Unable to locate')
       end
       waitrt?
+      @pet_count -= 1
```

Comment:
❓ Since this says pet boxes, is the assumption that boxes you pick up off the ground _won't_ be used for picking?

Reply:
You don't disarm during combat, only pick, so there's no chance you'll pick up a box and be able to disarm it.